server/src/main/java/keywhiz/ServiceModule.java

```diff
@@ -121,20 +125,34 @@ public ServiceModule(KeywhizConfig config, Environment environment) {
     return dbi;
   }
 
+  @Provides @Singleton ManagedDataSource writableDataSource(Environment environment, KeywhizConfig config) {
+    DataSourceFactory dataSourceFactory = config.getDataSourceFactory();
+    ManagedDataSource dataSource = dataSourceFactory.build(environment.metrics(), "postgres-writable");
+    environment.lifecycle().manage(dataSource);
+
+    return dataSource;
+  }
+
+  @Provides @Singleton
+  @Readonly ManagedDataSource readonlyDataSource(Environment environment, KeywhizConfig config) {
+    logger.debug("Creating read-only data source");
+    DataSourceFactory dataSourceFactory = config.getReadonlyDataSourceFactory();
+    ManagedDataSource dataSource = dataSourceFactory.build(environment.metrics(),
+        "postgres-readonly");
+    // TODO: do we need to do environment.lifecycle().manage(dataSource)?
```

Comment:
@sul3n3t will probably have a better answer but it seems like a good idea: http://dropwizard.github.io/dropwizard/manual/core.html#managed-objects